### PR TITLE
Sandraros/check demo regressions

### LINKED ISSUE #894: Unit Tests to avoid regression in existing demo programs
Proposing this issue as an open discussion.

I'd like to know your opinion about adding "unit" tests for the demo programs. That would imply:
- to store the XLSX files generated by the demo programs in the Web Repository (W3MI objects)
- to extract the code from the demonstration programs into new classes which can be called from the demo programs and from a test program
- to create a test program whose ABAP Unit test checks that the file generated by each demo program 

In the future, we could also use this feature to make sure that both "Writer" and "Huge Writer" generate the same result (currently we have many features which don't work in the Huge Writer), and the same for "Reader" and "Huge Reader". This 
[program](https://github.com/sapmentors/abap2xlsx/files/7577209/abap_program_to_compare_writer_and_huge_writer.txt) can be used as a starting point for discussion on Writer/Huge Writer, it runs 2 demos online and displays the diff, which works as explained in #891, and it also contains the test classes for them.

### PR DESCRIPTION
**Pull request by sandraros**
Fix #894

### CODE REVIEW

File: src/demos/zdemo_excel_checker.prog.abap [hunk 1]
```diff
@@ -0,0 +1,580 @@
...
+
+    DATA: lx TYPE REF TO cx_root.
+
+    TRY.
+
+        CASE ref_sscrfields->ucomm.
+
+          WHEN 'FC01'. " REFRESH
+
+            " restart the program completely so that to consider modification/recompile of any ZDEMO_EXCEL program
```
Comment (AndreaBorgia-Abo) [inline src/demos/zdemo_excel_checker.prog.abap]:
```suggestion
            " restart the program completely to account for modification/rebuild of any ZDEMO_EXCEL program
```

**Comment by AndreaBorgia-Abo** [comment]
I'll do the full review over the coming days, stay tuned.

**Comment by AndreaBorgia-Abo** [comment]
Review time...

**Comment by sandraros** [comment]
@AndreaBorgia-Abo I forgot to say that I was working on a [second simplest version](https://github.com/abap2xlsx/abap2xlsx/pull/940) to check demo regressions, whose advantage is to not change the demo programs. It's simply based on the Excel files created on the frontend by program `ZDEMO_EXCEL`. Could you say which solution you prefer?

**Comment by AndreaBorgia-Abo** [comment]
Sheesh, ended up using this PR as a guinea pig for a [possible abapGit bug](https://github.com/abapGit/abapGit/issues/5233#issuecomment-1004000633) ... I'll check the newer version first, then!

**Review by AndreaBorgia-Abo — CHANGES_REQUESTED** [review]
Should probably be closed in favour of the newer #940

**Comment by sandraros** [comment]
Closing in favor of other PR #940

**Pull request closed without merging**